### PR TITLE
Fixes #26535 - Fix Content reimport

### PR DESCRIPTION
**Pull request by jturel**
To test:

- enable some red hat repositories (not necessary, but it's a good test)
- Note how much content you have from the console and destroy them all

```
Katello::ProductContent.all.size
Katello::Content.all.size
Katello::Content.destroy_all
```
run `rake katello:reimport`

Check the counts again - they should be the same or similar if you had descrepencies.

You should be able to disable Red Hat repos without error

### CODE REVIEW

File: app/models/katello/content.rb [hunk 1]
```diff
@@ -37,17 +37,20 @@ def propagate_name_change?
...
+
         org.products.each do |product|
-          begin
-            product_json = Katello::Resources::Candlepin::Product.get(org.label,
-                                                                  product.cp_id,
-                                                                  %w(productContent)).first
-            product_content_attrs = product_json['productContent']
-            Katello::Glue::Candlepin::Product.import_product_content(product, product_content_attrs)
-          rescue RestClient::NotFound
+          product_json = cp_products.find { |prod| prod['id'] == product.cp_id }
```
Comment (parthaa) [inline app/models/katello/content.rb]: I am ok with this change. However I am wondering if you want to do a hash with id as key instead.
Its an overkill if there are only like 10 products per org, but if there are 100s hash might be better,

Comment (jturel) [reply app/models/katello/content.rb]: I think it's probably negligible since it's sandwiched in between an HTTP request and multiple database operations

Comment (parthaa) [reply app/models/katello/content.rb]: Hmm I saw this in one of the production databases
cp_products.count
=> 113
> o.products.count
=> 55

I think its worth making it a hash

Comment (jturel) [reply app/models/katello/content.rb]: updated!

**Comment by parthaa** [comment]
ack

**Comment by parthaa** [comment]
Works well .. Mind adding a test -> along the lines of -> https://github.com/Katello/katello/blob/master/test/models/rpm_test.rb#L162

File: test/models/content_test.rb [hunk 2]
```diff
@@ -20,5 +20,36 @@ def test_update_repo_name
 
       assert_includes repo.reload.name, "FOOBAR"
     end
+
+    def test_import_all
```
Comment (jturel) [inline test/models/content_test.rb]: @parthaa good call. here's a test

Comment (jturel) [reply test/models/content_test.rb]: Not a great test - but the ProductContentImporter is where the heavy lifting happens and it's well tested

**Comment by parthaa** [comment]
ack

**Pull request merged**